### PR TITLE
axial_point with basic functions 

### LINKED ISSUE #2: Add features and tests to `AxialPoint`
The `AxialPoint` type is currently very incomplete. It should implement all fitting traits from `cgmath` and overload many operators:

- overload many useful operators (including `Index`, required by `Array` anyway)
- implement `cgmath::{Zero, Array, MetricSpace, VectorSpace, InnerSpace}`

Everything should be done similar to [cgmath::Point2](http://bjz.github.io/cgmath/cgmath/struct.Point2.html).

Additionally multiple unit tests should be added to ensure correctness.

*Note*: this issue is similar to #1

### PR DESCRIPTION
**Pull request by RomanSchott**
closes #2

### CODE REVIEW

File: base/src/math/axial_point.rs [hunk 1]
```diff
@@ -51,3 +53,114 @@ impl fmt::Debug for AxialPoint {
...
 }
+
+/// ********************Basic Arithmetics************
+
+impl Add<AxialPoint> for AxialPoint {
+    type Output = AxialPoint;
+
+    /// adds two Points together similar to vectors
+    /// Returns an AxialPoint
+    fn add(self, _rhs: AxialPoint) -> AxialPoint {
```
Comment (jonas-schievink) [inline base/src/math/axial_point.rs]: Don't use leading `_` in variable names, this disables the `unused_variable` Lint (but this variable is used)

File: base/src/math/axial_point.rs [hunk 2]
```diff
@@ -51,3 +53,114 @@ impl fmt::Debug for AxialPoint {
...
+        }
+    }
+}
+
+impl Sub<AxialPoint> for AxialPoint {
+    type Output = AxialPoint;
+
+    /// substracts two Points similar to vectors
+    /// Returns an AxialPoint
+    fn sub(self, _rhs: AxialPoint) -> AxialPoint {
```
Comment (jonas-schievink) [inline base/src/math/axial_point.rs]: (same here)

File: base/src/math/axial_point.rs [hunk 3]
```diff
@@ -51,3 +53,114 @@ impl fmt::Debug for AxialPoint {
...
+    fn is_zero(&self) -> bool {
+        self.q == 0 && self.r == 0
+    }
+}
+
+/// ********************Index************
+
+impl Index<usize> for AxialPoint {
+    type Output = AxialType;
+    fn index<'a>(&'a self, index: usize) -> &'a AxialType {
```
Comment (jonas-schievink) [inline base/src/math/axial_point.rs]: Lifetime annotations should be unnecessary here

File: base/src/math/axial_point.rs [hunk 4]
```diff
@@ -51,3 +53,114 @@ impl fmt::Debug for AxialPoint {
...
+        match index {
+            0 => &self.q,
+            1 => &self.r,
+            _ => panic!("Index out of bounds!"),
+        }
+    }
+}
+
+impl IndexMut<usize> for AxialPoint {
+    fn index_mut<'a>(&'a mut self, index: usize) -> &'a mut AxialType {
```
Comment (jonas-schievink) [inline base/src/math/axial_point.rs]: Here, too (lifetimes shouldn't be necessary)

File: base/src/math/axial_point.rs [hunk 5]
```diff
@@ -51,3 +53,114 @@ impl fmt::Debug for AxialPoint {
...
+    fn sum(self) -> AxialType {
+        self.q + self.r
+    }
+
+    fn product(self) -> AxialType {
+        self.q * self.r
+    }
+
+    fn min(self) -> AxialType {
+        if self.q < self.r { self.q } else { self.r }
```
Comment (jonas-schievink) [inline base/src/math/axial_point.rs]: Prefer `std::cmp::min`

File: base/src/math/axial_point.rs [hunk 6]
```diff
@@ -51,3 +53,114 @@ impl fmt::Debug for AxialPoint {
...
+    }
+    fn max(self) -> AxialType {
+        if self.q < self.r { self.r } else { self.q }
```
Comment (jonas-schievink) [inline base/src/math/axial_point.rs]: (and `std::cmp::max` here)

File: base/src/math/axial_vector.rs [hunk 7]
```diff
@@ -54,3 +71,202 @@ impl fmt::Debug for AxialVector {
...
+    }
+}
+
+
+impl AddAssign<AxialVector> for AxialVector {
+
+    fn add_assign(&mut self, arg2: AxialVector) {
+        self.r += arg2.r;
+        self.q += arg2.q;
+
```
Comment (jonas-schievink) [inline base/src/math/axial_vector.rs]: Unnecessary blank line

**Comment by jonas-schievink** [comment]
There's a lot of unnecessary blank lines in here. Don't put empty lines after `{` or before `}`.

**Comment by jonas-schievink** [comment]
I recommend putting empty lines between items to separate them better.

File: base/src/math/axial_point.rs [hunk 8]
```diff
@@ -51,3 +49,150 @@ impl fmt::Debug for AxialPoint {
...
+/// ******************* Metric-Space************************
+impl MetricSpace for AxialPoint {
+    type Metric = f32;
+    fn distance2(self, other: AxialPoint) -> Self::Metric {
+        (((self.q - other.q) * (self.q - other.q)) +
+         ((self.r - other.r) * (self.r - other.r))) as f32
+    }
+    fn distance(self, other: AxialPoint) -> Self::Metric {
+        self.distance2(other).sqrt()
+    }
```
Comment (LukasKalbertodt) [inline base/src/math/axial_point.rs]: You don't need to implement this method as it is provided by the `MetrixSpace` trait already.

**Comment by LukasKalbertodt** [comment]
I think you have accidentally checked in the file from your partner group ( #25 ). Please remove it as this PR should just be about `AxialPoint`.

Also: your unit tests should be structured differently. There should be one test function for each feature or feature group. See #25 for a slightly longer explanation.

And ask me to fix your git history tomorrow in the morning. This should be done as early as possible.

**Comment by LukasKalbertodt** [comment]
replaced by #40

**Pull request closed without merging**